common/src/main/java/com/gregtechceu/gtceu/api/block/BlockLongDistancePipe.java

```diff
@@ -0,0 +1,104 @@
+package com.gregtechceu.gtceu.api.block;
+
+import com.gregtechceu.gtceu.api.pipenet.longdist.ILDEndpoint;
+import com.gregtechceu.gtceu.api.pipenet.longdist.LongDistanceNetwork;
+import com.gregtechceu.gtceu.api.pipenet.longdist.LongDistancePipeType;
+import gregtech.api.GregTechAPI;
+import gregtech.api.items.toolitem.ToolClasses;
+import net.minecraft.block.Block;
+import net.minecraft.block.material.Material;
+import net.minecraft.block.state.IBlockState;
+import net.minecraft.client.resources.I18n;
+import net.minecraft.client.util.ITooltipFlag;
+import net.minecraft.creativetab.CreativeTabs;
+import net.minecraft.entity.EntityLiving;
+import net.minecraft.entity.EntityLivingBase;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.NonNullList;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.world.IBlockAccess;
+import net.minecraft.world.World;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BlockLongDistancePipe extends Block {
+
+    private final LongDistancePipeType pipeType;
+
+    public BlockLongDistancePipe(LongDistancePipeType pipeType) {
+        super(Material.IRON);
+        this.pipeType = pipeType;
+        setTranslationKey("long_distance_" + pipeType.getName() + "_pipeline");
+        setCreativeTab(GregTechAPI.TAB_GREGTECH);
+        setHarvestLevel(ToolClasses.WRENCH, 1);
+    }
+
+    @Override
+    public void onBlockPlacedBy(@Nonnull World worldIn, @Nonnull BlockPos pos, @Nonnull IBlockState state, @Nonnull EntityLivingBase placer, @Nonnull ItemStack stack) {
+        super.onBlockPlacedBy(worldIn, pos, state, placer, stack);
+        if (worldIn.isRemote) return;
+        // first find all neighbouring networks
+        List<LongDistanceNetwork> networks = new ArrayList<>();
+        BlockPos.PooledMutableBlockPos offsetPos = BlockPos.PooledMutableBlockPos.retain();
+        for (EnumFacing facing : EnumFacing.VALUES) {
+            offsetPos.setPos(pos).move(facing);
+            LongDistanceNetwork network = LongDistanceNetwork.get(worldIn, offsetPos);
```

Comment:
The first file I started reviewing was absolutely devastating. There was even `EnumFacing`, not to mention compiling, and you certainly didn't even use IDEA for coding. Under such circumstances, you still ask for review?

Reply:
Kila.
I spesifically asked them to **just** copy over all the classes

Reply:
Ah fine, I see